20170613-mock/code/ggee/guessing/game.py

```diff
@@ -47,14 +47,14 @@ def think_of_a_number(level):
 
 def play(level):
     # type: (int) -> None
-    print(T("Level {level}: 0 to {max_num}").format(
-        level=level, max_num=max_number(level)))
+    print(T("Level {level}: 0 to {max_num}".format(
+        level=level, max_num=max_number(level))))
```

Comment:
Actually, when the implementation of `T` is not as stupid as I did it, the original is the correct line. Here how it suppose to work:

Some pre-processor should run on this code and collect all the `T("...")` lines and put all the strings into a `.pot` file (portable object template). Then human translators (with the aid of a dedicated s/w) will translate each string to their target languages, creating a set of `.po` files (portable object). Our s/w will pick the correct file according the the system's local, and use the string in T("...") as a key to find the translated string in that file, and use it.

So as a result you need to keep the `{...}` in the call to `T` and the translator need to keep them (although she might need to move them around, if the target language requires it). If we will not do this, there might be infinite strings to translate.

I appreciate that a machine translate might muck things here, but consider the case where we want to call the translate service only once and cache the results.